ballerina/http_response.bal

```diff
@@ -562,18 +562,34 @@ public class Response {
         return payload;
     }
 
-    isolated function getStatusCodeResponseBindingError(string reasonPhrase, boolean generalError) returns ClientError {
+    isolated function getStatusCodeResponseBindingError(string reasonPhrase) returns ClientError {
+        map<string[]> headers = getHeaders(self);
+        anydata|error payload = getPayload(self);
+        int statusCode = self.statusCode;
+        if payload is error {
+            if payload is NoContentError {
+                return createStatusCodeResponseBindingError(statusCode, reasonPhrase, headers);
+            }
+            return error PayloadBindingClientError("http:StatusCodeBindingError creation failed: " + statusCode.toString() +
+                " response payload extraction failed", payload);
+        } else {
+            return createStatusCodeResponseBindingError(statusCode, reasonPhrase, headers, payload);
+        }
+    }
+
+    isolated function getStatusCodeResponseDataBindingError(string reasonPhrase, boolean fromDefaultStatusCodeMapping,
```

Comment:
Ditto. Refer this comment [1]

[1] - https://github.com/ballerina-platform/module-ballerina-http/pull/2002/files#r1616543125

Reply:
Fixed via [3ed8b32](https://github.com/ballerina-platform/module-ballerina-http/pull/2002/commits/3ed8b32953953b57dc0074a09681ce99e755e232)